src/main/java/com/dataStructures/HashTable.java

```diff
@@ -0,0 +1,45 @@
+package src.main.java.com.dataStructures;
+
+/**
+ * Hash table without repetitions
+ * @author Rick
+ * @param <E> Data type
+ * @param <T> Key type
+ */
+public class HashTable <E,T> {
+    
+    private E[] data_table;
+
+    public HashTable(int table_size) {
+        data_table = (E[])(new Object[table_size]);
+    }
+
+    public HashTable() {
+        this(50);
+    }
+
+    private int hash(String key){
+        int sum = 0;
+        for (char c : key.toCharArray()) {
+            sum += (int)c;
+        }
+        return sum % this.data_table.length;
+    }
+    
+    public void add(E data, T key){
```

Comment:
The convention for any hash is for the key to be the first parameter, so this signature would be better if it were:
```java
public void add(T key, E data)
```